packages/nimble/R/genCpp_sizeProcessing.R

```diff
@@ -566,83 +568,103 @@ sizeRep <- function(code, symTab, typeEnv) {
+        code$toEigenize <- "maybe"
+        code$nDim <- 0
+        ##      if(!(className %in% names(nfProc$neededTypes))) nfProc$neededTypes[[className]] <- newSym
+    }##stop('Error in sizeNewNimbleList: listGenerator not found in parentST', call. = FALSE)
     
-    ## embed RnewExprs in a '{' expression
-    if(length(RnewExprs) != 0) {
-      RbracketNewExprs <- quote(after({}))
-      RbracketNewExprs[[2]][2:(length(RnewExprs) + 1)] <- RnewExprs
-      bracketNewExprs <- RparseTree2ExprClasses(RbracketNewExprs)
-      ## Need to install assignment target in symTab if necessary so that it
-      ## will be there for recursion in the following step
-      assignmentTarget <- code$caller$args[[1]]
-      if(assignmentTarget$isName) {
-        if(!symTab$symbolExists(assignmentTarget$name, TRUE)) {
-          symTab$addSymbol(symbolNimbleList(name = assignmentTarget$name, type = code$type, nlProc = code$sizeExprs$nlProc))
+    asserts <- list()
+    asserts <- c(asserts, recurseSetSizes(code, symTab, typeEnv, useArgs = c(TRUE, rep(FALSE, length(code$args)-1))))
+    if(!(code$caller$name %in% assignmentOperators)){
+        intermediateAsserts <- sizeInsertIntermediate(code$caller, code$callerArgID, symTab, typeEnv)
+        ## intermediateAsserts can potentially have size setting stuff from sizeAssignAfterRecursing.
+        ## Not sure if that would ever happen in this context, but to be safe we'll use last element as the actual intermediate assignment
+        ## Embed the intermediate assignment in a '{' (so insertAssertions will recurse on it) and recurse on it
+        numIntermAsserts <- length(intermediateAsserts)
+        bracketedIntermAssert <- newBracketExpr(intermediateAsserts[numIntermAsserts])
+        exprClasses_setSizes(bracketedIntermAssert, symTab, typeEnv)
+        intermediateAsserts[[numIntermAsserts]] <- bracketedIntermAssert
+        asserts <- c(asserts, intermediateAsserts)
+        return(asserts)
+    }
+    if(length(code$args)>1){   
+        RnewExprs <- list()
+        newExprs <- list()
+        RnfVarExprs <- list()
+        nfVarExprs <- list()
+        exprCounter <- 1
+        originalCode <- code 
+        listElements <- listST$nlProc$symTab$getSymbolNames() ##getSymbolObjects()
+        RlistNameExpr <- nimbleGeneralParseDeparse(originalCode$caller$args[[1]])    
+        for(i in seq_along(listElements)) {
+            thisVarName <- listElements[i]
+            ##      if(!inherits(originalCode$args[[i+1]], 'exprClass') ||  (originalCode$args[[i+1]]$name != "")){  ## skip first arg, which will be name of nlDef, then check if value is ""
+            if(!is.null(originalCode$args[[thisVarName]])) {
+                if(!inherits(originalCode$args[[thisVarName]], 'exprClass') ||  (originalCode$args[[thisVarName]]$name != "")){  ## skip first arg, which will be name of nlDef, then check if value is ""
+                    ## nfVar(A, 'x') for whichever element name it's on ('x')
```

Comment:
Is this the beginning of a sentence? Some nice style rules are:
- Start sentences with capital letters.
- Finish sentences with periods.
- Be grammatical.
This may seem pedantic until you have to deal with a truncated comment (due to merge conflict etc.) and you can't determine whether you've read all the information.